### PR TITLE
`Development`: Replace ngx-slider with angular material slider

### PR DESCRIPTION
**Pull request by egekurt123**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Description

Solves #8105 

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Students
- 1 Exam

1. Log in to Artemis
2. Navigate to Course Administration
3. Create an Exam and conduct it
4. Participate in the exam with a student account
5. Make multiple submissions of exercises within the exam.
6. Submit the exam
7. Log back in with an Instructor
8. Navigate to Student Exams and select the exam you submitted with the student account
9. Click on the "Exam timeline" button
10. Make sure that the slider works as expected

[Example timeline on ts5](https://artemis-test5.artemis.cit.tum.de/course-management/10/exams/29/student-exams/54/exam-timeline)

#### Performance Review
- [x] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Screenshots

Screenshot 2024-03-09 at 23 27 07

## Summary by CodeRabbit

- **New Features**
	- Introduced a custom slider component for managing student exams, enhancing user interaction.
- **Refactor**
	- Replaced `NgxSliderModule` with `MatSliderModule` for improved performance and aesthetics in exam management components.
- **Style**
	- Customized the appearance of the new slider component to align with the application's design.

### CODE REVIEW

**Review by beyzaaltuntas — APPROVED** [review]
Tested with TS5, works as expected

File: src/main/webapp/app/exam/manage/student-exams/student-exam-timeline/student-exam-timeline.component.scss [hunk 1]
```diff
@@ -0,0 +1,49 @@
+::ng-deep {
```
Comment (basak-akan) [inline src/main/webapp/app/exam/manage/student-exams/student-exam-timeline/student-exam-timeline.component.scss:1]: Please see our [client guideline](https://docs.artemis.cit.tum.de/dev/guidelines/client) for ng-deep usage. 
It is advised to use a ` :host` in front of the command.

**Review by basak-akan — COMMENTED** [review]
Tested during the testing session on TS5, worked well. 
Code lgtm, I just have one comment.

**Review by basak-akan — APPROVED** [review]
Thanks for addressing my feedback

**Review by tobias-lippert — COMMENTED** [review]
Tested on ts5. Everything still works with the new slider.
The only thing I've noticed is that the old one showed a tooltip of the time when hovering over the tick on the slider. So you could see the timestamp without actually selecting it.
Do you think that's feasible with the new slider?

**Comment by egekurt123** [comment]
> Tested on ts5. Everything still works with the new slider. The only thing I've noticed is that the old one showed a tooltip of the time when hovering over the tick on the slider. So you could see the timestamp without actually selecting it. Do you think that's feasible with the new slider?

Thanks for your remark, i took a detailed look into this: 
This would require a workaround implemented in a wrapper class that includes a new ticking option with a new array, position calculations for hovering, event listeners for those hoverings, and updates when the value of the slider is changed.
As it is implemented in ngx-slider: https://github.com/angular-slider/ngx-slider

**Review by tobias-lippert — APPROVED** [review]
As you described it, this change is probably out of scope.
Approve then.

**Review by laurenzfb — APPROVED** [review]
tested on ts5. slider works

**Review by beyzaaltuntas — APPROVED** [review]
Reapprove.

**Pull request merged**